File: hw/chavez/u3/hw17OPPConceptZoo/Python/OPPConceptsZoo/controller/animal_controller.py
from model.cow import Cow
from model.pig import Pig
from model.sheep import Sheep
from model.chicken import Chicken


class AnimalController:

    def create_animal(self, animal_type, id, breed, born_on, weight):

        if animal_type == "Cow":
            return Cow(id, breed, born_on, weight)

        elif animal_type == "Pig":
            return Pig(id, breed, born_on, weight)

        elif animal_type == "Sheep":
            return Sheep(id, breed, born_on, weight)

        elif animal_type == "Chicken":
            return Chicken(id, breed, born_on, weight)

        return None

File: hw/chavez/u3/hw17OPPConceptZoo/JavaScript/OPPConceptsZoo/controller/AnimalController.js
import Cow from "../model/Cow.js";
import Pig from "../model/Pig.js";
import Chicken from "../model/Chicken.js";
import Sheep from "../model/Sheep.js";

export default class AnimalController {

    createAnimal(type, id, breed, bornOn, weight) {

        switch (type) {

            case "Cow":
                return new Cow(
                    id,
                    breed,
                    bornOn,
                    weight
                );

            case "Pig":
                return new Pig(
                    id,
                    breed,
                    bornOn,
                    weight
                );

            case "Chicken":
                return new Chicken(
                    id,
                    breed,
                    bornOn,
                    weight
                );

            case "Sheep":
                return new Sheep(
                    id,
                    breed,
                    bornOn,
                    weight
                );

            default:
                return null;
        }
    }

}